portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -69,20 +73,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
       // Get the input from the form
-      String comment = request.getParameter("comment");
-      Entity commentEntity = new Entity("Comment");
-      commentEntity.setProperty("comment", comment);
-      datastore.put(commentEntity);
+      if (!request.getParameter(PROPERTY_COMMENT).isEmpty() && request.getParameter(PROPERTY_COMMENT) != null) {
+        String comment = request.getParameter(PROPERTY_COMMENT);
+        String name = getParameterOrDefault(request, PROPERTY_USERNAME, "anonymous");
+        Entity commentEntity = new Entity(ENTITY_COMMENT);
+        commentEntity.setProperty(PROPERTY_COMMENT, comment);
+        commentEntity.setProperty(PROPERTY_USERNAME, name);
+        datastore.put(commentEntity);
+      }
       
       // Redirect back to index page
       response.sendRedirect("/index.html");
   }
 
-
   private String convertToJsonUsingGson(List<String> comments) {
     return new Gson().toJson(comments);
   }
 
+  private String getParameterOrDefault(HttpServletRequest request, String parameter, String def) {
+    if (!request.getParameter(parameter).isEmpty() && request.getParameter(parameter) != null) {
+        return request.getParameter(parameter);
+      }
+    return def;
+  }
+
+  private String getPropertyOrDefault(Entity entity, String property, String def) {
+    if (entity.getProperty(property) == null) {
+        return def;
+      }
+    return (String) entity.getProperty(property);
```

Comment:
nit: same here, consider storing to a local

also I noticed handling for the empty string differs between `#getParameterOrDefault` and `#getPropertyOrDefault` -- one method isn't necessarily better than the other, but it's always good to be consistent